test/integration/dns_test.go

```diff
@@ -187,20 +187,29 @@ func TestDNS(t *testing.T) {
 			expect:          []*net.IP{&headlessIP},
 		},
 		{ // specific port of a headless service
-			dnsQuestionName: "unknown-port-2345.e1.headless.default.svc.cluster.local.",
+			dnsQuestionName: "_http._tcp.headless.default.svc.cluster.local.",
```

Comment:
It looks like none of these exercise the `_<protocol>.<svc>.<ns>.svc` case (wildcard port match), so there should probably be such a test here.  Furthermore, it looks like none of these test the `.endpoints` queries, and that seems like it would be valuable to test here as well.